### PR TITLE
[TEST](indice_convolution_backward_filter):add api check cases for indice_convolution_backward_filter

### PR DESCRIPTION
**Pull request by Shin-Wang**
…dice_convolution_backward_filter

Thanks for your contribution and we appreciate it a lot. 

## 1. Motivation

Please describe your motivation and the goal you want to achieve through this pull request.

## 2. Modification

Please briefly describe what modification is made in this pull request, and indicate where to make the modification.

## 3. Test Report

If you want to know how to do operator testing, you can see [GTest-User-Guide-zh](https://github.com/Cambricon/mlu-ops/blob/master/docs/GTest-User-Guide-zh.md).

#### 3.1.1 Accuracy Acceptance Standard

For static threshold standard details, see: [MLU-OPS Accuracy Acceptance Standard](https://github.com/Cambricon/mlu-ops/blob/master/docs/MLU-OPS-Accuracy-Acceptance-Standard.md).

- [ ] diff1: diff1 <= 3e-3
- [ ] diff2: diff2 <= 3e-3

#### 3.1.2 Operator Scheme checklist

|     No.        |                 Details              |            Check Results             |
|----------------|--------------------------------------|--------------------------------------|
|        1       |Supported hardware                    |             MLU370<br>MLU590         |
|        2       |Job types                             |          block <br> U1 <br> U4       |
|        3       |Layouts                               |          NHWC 、NCHW、ARRAY etc      |
|        4       |Whether multi-dimensions are supported|                                      |
|        5       |Whether element zero is supported     |                                      |
|        6       |Data type(half/float)                 |           half / float etc           |
|        7       |Whether there is size limit           |                                      |

#### 3.1.3 New Feature Test

If you have checked the following items, please tick the relevant box.

- [ ] Data type test
- [ ] Multidimensional tensor test
- [ ] Layout test
- [ ] Different size/integer remainder end segment/alignment misalignment test
- [ ] Zero dimensional tensor test/zero element test
- [ ] stability test
- [ ] Multiple platform test
- [ ] Gen_case module test
- [ ] Nan/INF tests 
- [ ] Bug fix tests
- [ ] For memory leak check details, see[GTest-User-Guide-zh](https://github.com/Cambricon/mlu-ops/blob/master/docs/GTest-User-Guide-zh.md).
- [ ] For code coverage check details, see: [GTest-User-Guide-zh](https://github.com/Cambricon/mlu-ops/blob/master/docs/GTest-User-Guide-zh.md).
- [ ] For I/O calculation efficiency check details, see: [MLU-OPS Performance Acceptance Standard](https://github.com/Cambricon/mlu-ops/blob/master/docs/MLU-OPS-Performance-Acceptance-Standard.md).

#### 3.1.4 Parameter Check

When a new operator is submitted, the test points are given and the test results are stated.

|                   Test Point                    | Acceptance Standard | Test Result (Error Message) |
| ----------------------------------------------- | --------------------| --------------------------- |
| Whether it conforms to the operator restriction |     Normal error    |                             |
| Whether illegal parameters are passed           |     Normal error    |                             |

### 3.2 Accuracy Test

For the cases used in the New Feature Test section, the features and the number of cases are recorded here. When multiple operations are tested, multiple tables are needed to include details of these operations.

Operation:

|Test Point           | Description                      | Quantity |  Comment |
|----------           |----------------------------------|----------|  --------|
|Data type test       |half/float/int8                   |          |          |
|Mult-tensor test     |Supports 1-8 dims                 |          |          |
|Layout test          |Supports NCHW/NHWC                |          |          |
|Zero element test    |Whether to support this test      |          |          |
|Stability test       |--gtest_repeat=NUM<br>--thread=NUM|          |          |
|Mult-platform test   |MLU370/MLU590                     |          |          |
|Nan/INF test         |Whether to support this test      |          |          |

### 3.3 Performance Test

See [MLU-OPS Performance Acceptance Standard](https://github.com/Cambricon/mlu-ops/blob/master/docs/MLU-OPS-Performance-Acceptance-Standard.md) for details.

Platform：MLU370

|Operation|Mlu_hardware_time(us)|Mlu_interface_time(us)|Mlu_io_efficiency|Mlu_compute_efficiency|Mlu_workwpace_size(Bytes)|Data_type|Shape|
|-------|----|----|-----|----|----|----|-----|
|op_name|    |    |     |    |    |    |     |
|op_name|    |    |     |    |    |    |     |

Platform：MLU590

|Operation|Mlu_hardware_time(us)|Mlu_interface_time(us)|Mlu_io_efficiency|Mlu_compute_efficiency|Mlu_workwpace_size(Bytes)|Data_type|Shape|
|-------|----|----|----|----|----|----|-----|
|op_name|    |    |    |    |    |    |     |
|op_name|    |    |    |    |    |    |     |

### 3.4 Summary Analysis

Please give a brief overview here, if you want to note and summarize the content.

### CODE REVIEW

**Review by FRYao — APPROVED** [review]

File: bangc-ops/test/mlu_op_gtest/api_gtest/src/gtest/indice_convolution_backward_filter/indice_convolution_backward_filter.cpp [hunk 1]
```diff
@@ -0,0 +1,360 @@
...
+      if (features_desc) {
+        GTEST_CHECK(
+            CNRT_RET_SUCCESS ==
+            cnrtMalloc(&features_, mluOpGetTensorElementNum(features_desc_) *
+                                       mluOpDataTypeBytes(MLUOP_DTYPE_FLOAT)));
+      } else {
+        GTEST_CHECK(
+            CNRT_RET_SUCCESS ==
+            cnrtMalloc(&features_, 64 * mluOpDataTypeBytes(MLUOP_DTYPE_FLOAT)));
+      }
```
Comment (alex-xuwenming) [inline bangc-ops/test/mlu_op_gtest/api_gtest/src/gtest/indice_convolution_backward_filter/indice_convolution_backward_filter.cpp:63]: 这是什么意思？为啥为空的时候需要默认申请空间？

**Comment by Shin-Wang** [comment]
测试报告：检查api ptr不为空；检查api 参数异常情况

**Comment by Shin-Wang** [comment]
```
[----------] 11 tests from indice_convolution_backward_filter
[ RUN      ] indice_convolution_backward_filter.BAD_PARAM_handle_null
[2023-2-22 11:40:23] [MLUOP] [Error]:[mluOpIndiceConvolutionBackwardFilter] Check failed: handle != nullptr. 
[       OK ] indice_convolution_backward_filter.BAD_PARAM_handle_null (311 ms)
[ RUN      ] indice_convolution_backward_filter.BAD_PARAM_features_desc_null
[2023-2-22 11:40:23] [MLUOP] [Error]:[mluOpIndiceConvolutionBackwardFilter] Check failed: features_desc != nullptr. 
[       OK ] indice_convolution_backward_filter.BAD_PARAM_features_desc_null (0 ms)
[ RUN      ] indice_convolution_backward_filter.BAD_PARAM_features_null
[2023-2-22 11:40:23] [MLUOP] [Error]:[mluOpIndiceConvolutionBackwardFilter] Check failed: features != nullptr. 
[       OK ] indice_convolution_backward_filter.BAD_PARAM_features_null (0 ms)
[ RUN      ] indice_convolution_backward_filter.BAD_PARAM_output_grad_desc_null
[2023-2-22 11:40:23] [MLUOP] [Error]:[mluOpIndiceConvolutionBackwardFilter] Check failed: output_grad_desc != nullptr. 
[       OK ] indice_convolution_backward_filter.BAD_PARAM_output_grad_desc_null (0 ms)
[ RUN      ] indice_convolution_backward_filter.BAD_PARAM_output_grad_null
[2023-2-22 11:40:23] [MLUOP] [Error]:[mluOpIndiceConvolutionBackwardFilter] Check failed: output_grad != nullptr. 
[       OK ] indice_convolution_backward_filter.BAD_PARAM_output_grad_null (0 ms)
[ RUN      ] indice_convolution_backward_filter.BAD_PARAM_indice_pairs_desc_null
[2023-2-22 11:40:23] [MLUOP] [Error]:[mluOpIndiceConvolutionBackwardFilter] Check failed: indice_pairs_desc != nullptr. 
[       OK ] indice_convolution_backward_filter.BAD_PARAM_indice_pairs_desc_null (0 ms)
[ RUN      ] indice_convolution_backward_filter.BAD_PARAM_indice_pairs_null
[2023-2-22 11:40:23] [MLUOP] [Error]:[mluOpIndiceConvolutionBackwardFilter] Check failed: indice_pairs != nullptr. 
[       OK ] indice_convolution_backward_filter.BAD_PARAM_indice_pairs_null (0 ms)
[ RUN      ] indice_convolution_backward_filter.BAD_PARAM_filters_grad_desc_null
[2023-2-22 11:40:23] [MLUOP] [Error]:[mluOpIndiceConvolutionBackwardFilter] Check failed: filters_grad_desc != nullptr. 
[       OK ] indice_convolution_backward_filter.BAD_PARAM_filters_grad_desc_null (0 ms)
[ RUN      ] indice_convolution_backward_filter.BAD_PARAM_filters_grad_null
[2023-2-22 11:40:23] [MLUOP] [Error]:[mluOpIndiceConvolutionBackwardFilter] Check failed: filters_grad != nullptr. 
[       OK ] indice_convolution_backward_filter.BAD_PARAM_filters_grad_null (0 ms)
[ RUN      ] indice_convolution_backward_filter.BAD_PARAM_indice_num_null
[2023-2-22 11:40:23] [MLUOP] [Error]:[mluOpIndiceConvolutionBackwardFilter] Check failed: indice_num != nullptr. 
[       OK ] indice_convolution_backward_filter.BAD_PARAM_indice_num_null (0 ms)
[ RUN      ] indice_convolution_backward_filter.BAD_PARAM_workspace_null
[2023-2-22 11:40:23] [MLUOP] [Error]:[mluOpIndiceConvolutionBackwardFilter] Check failed: workspace != nullptr. 
[       OK ] indice_convolution_backward_filter.BAD_PARAM_workspace_null (0 ms)
[----------] 11 tests from indice_convolution_backward_filter (311 ms total)

```

**Comment by Shin-Wang** [comment]
```
[----------] 11 tests from indice_convolution_backward_filter
[ RUN      ] indice_convolution_backward_filter.BAD_PARAM_handle_null
[2023-2-22 11:40:23] [MLUOP] [Error]:[mluOpIndiceConvolutionBackwardFilter] Check failed: handle != nullptr. 
[       OK ] indice_convolution_backward_filter.BAD_PARAM_handle_null (311 ms)
[ RUN      ] indice_convolution_backward_filter.BAD_PARAM_features_desc_null
[2023-2-22 11:40:23] [MLUOP] [Error]:[mluOpIndiceConvolutionBackwardFilter] Check failed: features_desc != nullptr. 
[       OK ] indice_convolution_backward_filter.BAD_PARAM_features_desc_null (0 ms)
[ RUN      ] indice_convolution_backward_filter.BAD_PARAM_features_null
[2023-2-22 11:40:23] [MLUOP] [Error]:[mluOpIndiceConvolutionBackwardFilter] Check failed: features != nullptr. 
[       OK ] indice_convolution_backward_filter.BAD_PARAM_features_null (0 ms)
[ RUN      ] indice_convolution_backward_filter.BAD_PARAM_output_grad_desc_null
[2023-2-22 11:40:23] [MLUOP] [Error]:[mluOpIndiceConvolutionBackwardFilter] Check failed: output_grad_desc != nullptr. 
[       OK ] indice_convolution_backward_filter.BAD_PARAM_output_grad_desc_null (0 ms)
[ RUN      ] indice_convolution_backward_filter.BAD_PARAM_output_grad_null
[2023-2-22 11:40:23] [MLUOP] [Error]:[mluOpIndiceConvolutionBackwardFilter] Check failed: output_grad != nullptr. 
[       OK ] indice_convolution_backward_filter.BAD_PARAM_output_grad_null (0 ms)
[ RUN      ] indice_convolution_backward_filter.BAD_PARAM_indice_pairs_desc_null
[2023-2-22 11:40:23] [MLUOP] [Error]:[mluOpIndiceConvolutionBackwardFilter] Check failed: indice_pairs_desc != nullptr. 
[       OK ] indice_convolution_backward_filter.BAD_PARAM_indice_pairs_desc_null (0 ms)
[ RUN      ] indice_convolution_backward_filter.BAD_PARAM_indice_pairs_null
[2023-2-22 11:40:23] [MLUOP] [Error]:[mluOpIndiceConvolutionBackwardFilter] Check failed: indice_pairs != nullptr. 
[       OK ] indice_convolution_backward_filter.BAD_PARAM_indice_pairs_null (0 ms)
[ RUN      ] indice_convolution_backward_filter.BAD_PARAM_filters_grad_desc_null
[2023-2-22 11:40:23] [MLUOP] [Error]:[mluOpIndiceConvolutionBackwardFilter] Check failed: filters_grad_desc != nullptr. 
[       OK ] indice_convolution_backward_filter.BAD_PARAM_filters_grad_desc_null (0 ms)
[ RUN      ] indice_convolution_backward_filter.BAD_PARAM_filters_grad_null
[2023-2-22 11:40:23] [MLUOP] [Error]:[mluOpIndiceConvolutionBackwardFilter] Check failed: filters_grad != nullptr. 
[       OK ] indice_convolution_backward_filter.BAD_PARAM_filters_grad_null (0 ms)
[ RUN      ] indice_convolution_backward_filter.BAD_PARAM_indice_num_null
[2023-2-22 11:40:23] [MLUOP] [Error]:[mluOpIndiceConvolutionBackwardFilter] Check failed: indice_num != nullptr. 
[       OK ] indice_convolution_backward_filter.BAD_PARAM_indice_num_null (0 ms)
[ RUN      ] indice_convolution_backward_filter.BAD_PARAM_workspace_null
[2023-2-22 11:40:23] [MLUOP] [Error]:[mluOpIndiceConvolutionBackwardFilter] Check failed: workspace != nullptr. 
[       OK ] indice_convolution_backward_filter.BAD_PARAM_workspace_null (0 ms)
[----------] 11 tests from indice_convolution_backward_filter (311 ms total)

```

**Comment by Shin-Wang** [comment]
```
[----------] 2 tests from zero_element_1/indice_convolution_backward_filter_general
[ RUN      ] zero_element_1/indice_convolution_backward_filter_general.negative/0
[       OK ] zero_element_1/indice_convolution_backward_filter_general.negative/0 (0 ms)
[ RUN      ] zero_element_1/indice_convolution_backward_filter_general.negative/1
[       OK ] zero_element_1/indice_convolution_backward_filter_general.negative/1 (0 ms)
[----------] 2 tests from zero_element_1/indice_convolution_backward_filter_general (0 ms total)

[----------] 2 tests from zero_element_2/indice_convolution_backward_filter_general
[ RUN      ] zero_element_2/indice_convolution_backward_filter_general.negative/0
[       OK ] zero_element_2/indice_convolution_backward_filter_general.negative/0 (0 ms)
[ RUN      ] zero_element_2/indice_convolution_backward_filter_general.negative/1
[       OK ] zero_element_2/indice_convolution_backward_filter_general.negative/1 (0 ms)
[----------] 2 tests from zero_element_2/indice_convolution_backward_filter_general (0 ms total)

[----------] 2 tests from zero_element_3/indice_convolution_backward_filter_general
[ RUN      ] zero_element_3/indice_convolution_backward_filter_general.negative/0
[       OK ] zero_element_3/indice_convolution_backward_filter_general.negative/0 (0 ms)
[ RUN      ] zero_element_3/indice_convolution_backward_filter_general.negative/1
[       OK ] zero_element_3/indice_convolution_backward_filter_general.negative/1 (0 ms)
[----------] 2 tests from zero_element_3/indice_convolution_backward_filter_general (0 ms total)

[----------] 2 tests from bad_unsupport_dtype/indice_convolution_backward_filter_general
[ RUN      ] bad_unsupport_dtype/indice_convolution_backward_filter_general.negative/0
[       OK ] bad_unsupport_dtype/indice_convolution_backward_filter_general.negative/0 (0 ms)
[ RUN      ] bad_unsupport_dtype/indice_convolution_backward_filter_general.negative/1
[2023-2-22 11:40:23] [MLUOP] [Error]:[mluOpGetIndiceConvolutionBackwardFilterWorkspaceSize] The data type of features_desc, output_grad_desc and filters_grad_desc should be the same and the three should be either half or float. But now the data types are DTYPE_COMPLEX_FLOAT-DTYPE_COMPLEX_FLOAT-DTYPE_COMPLEX_FLOAT.
[       OK ] bad_unsupport_dtype/indice_convolution_backward_filter_general.negative/1 (0 ms)
[----------] 2 tests from bad_unsupport_dtype/indice_convolution_backward_filter_general (0 ms total)

[----------] 8 tests from bad_features_dim_shape_dtype/indice_convolution_backward_filter_general
[ RUN      ] bad_features_dim_shape_dtype/indice_convolution_backward_filter_general.negative/0
[       OK ] bad_features_dim_shape_dtype/indice_convolution_backward_filter_general.negative/0 (0 ms)
[ RUN      ] bad_features_dim_shape_dtype/indice_convolution_backward_filter_general.negative/1
[2023-2-22 11:40:23] [MLUOP] [Error]:[mluOpGetIndiceConvolutionBackwardFilterWorkspaceSize] The data type of features_desc, output_grad_desc and filters_grad_desc should be the same and the three should be either half or float. But now the data types are DTYPE_HALF-DTYPE_FLOAT-DTYPE_FLOAT.
[       OK ] bad_features_dim_shape_dtype/indice_convolution_backward_filter_general.negative/1 (0 ms)
[ RUN      ] bad_features_dim_shape_dtype/indice_convolution_backward_filter_general.negative/2
[       OK ] bad_features_dim_shape_dtype/indice_convolution_backward_filter_general.negative/2 (0 ms)
[ RUN      ] bad_features_dim_shape_dtype/indice_convolution_backward_filter_general.negative/3
[2023-2-22 11:40:23] [MLUOP] [Error]:[mluOpGetIndiceConvolutionBackwardFilterWorkspaceSize] Shape check failed! Now the shapes are features_desc[3,5,1], output_grad_desc[3,3], indice_pairs_desc[9,2,3], and filters_grad_desc[3,3,5,3].
[       OK ] bad_features_dim_shape_dtype/indice_convolution_backward_filter_general.negative/3 (0 ms)
[ RUN      ] bad_features_dim_shape_dtype/indice_convolution_backward_filter_general.negative/4
[       OK ] bad_features_dim_shape_dtype/indice_convolution_backward_filter_general.negative/4 (0 ms)
[ RUN      ] bad_features_dim_shape_dtype/indice_convolution_backward_filter_general.negative/5
[2023-2-22 11:40:23] [MLUOP] [Error]:[mluOpGetIndiceConvolutionBackwardFilterWorkspaceSize] Shape check failed! Now the shapes are features_desc[3,15], output_grad_desc[3,3], indice_pairs_desc[9,2,3], and filters_grad_desc[3,3,5,3].
[       OK ] bad_features_dim_shape_dtype/indice_convolution_backward_filter_general.negative/5 (0 ms)
[ RUN      ] bad_features_dim_shape_dtype/indice_convolution_backward_filter_general.negative/6
[       OK ] bad_features_dim_shape_dtype/indice_convolution_backward_filter_general.negative/6 (0 ms)
[ RUN      ] bad_features_dim_shape_dtype/indice_convolution_backward_filter_general.negative/7
[2023-2-22 11:40:23] [MLUOP] [Error]:[mluOpGetIndiceConvolutionBackwardFilterWorkspaceSize] Shape check failed! Now the shapes are features_desc[13,3], output_grad_desc[3,3], indice_pairs_desc[9,2,3], and filters_grad_desc[3,3,5,3].
[       OK ] bad_features_dim_shape_dtype/indice_convolution_backward_filter_general.negative/7 (0 ms)
[----------] 8 tests from bad_features_dim_shape_dtype/indice_convolution_backward_filter_general (0 ms total)

[----------] 8 tests from bad_indice_pairs_dim_shape_dtype/indice_convolution_backward_filter_general
[ RUN      ] bad_indice_pairs_dim_shape_dtype/indice_convolution_backward_filter_general.negative/0
[       OK ] bad_indice_pairs_dim_shape_dtype/indice_convolution_backward_filter_general.negative/0 (0 ms)
[ RUN      ] bad_indice_pairs_dim_shape_dtype/indice_convolution_backward_filter_general.negative/1
[2023-2-22 11:40:23] [MLUOP] [Error]:[mluOpGetIndiceConvolutionBackwardFilterWorkspaceSize] indice_pairs_desc only supports data type int32. But now the data type is DTYPE_INT64.
[       OK ] bad_indice_pairs_dim_shape_dtype/indice_convolution_backward_filter_general.negative/1 (0 ms)
[ RUN      ] bad_indice_pairs_dim_shape_dtype/indice_convolution_backward_filter_general.negative/2
[       OK ] bad_indice_pairs_dim_shape_dtype/indice_convolution_backward_filter_general.negative/2 (0 ms)
[ RUN      ] bad_indice_pairs_dim_shape_dtype/indice_convolution_backward_filter_general.negative/3
[2023-2-22 11:40:23] [MLUOP] [Error]:[mluOpGetIndiceConvolutionBackwardFilterWorkspaceSize] Shape check failed! Now the shapes are features_desc[3,5], output_grad_desc[3,3], indice_pairs_desc[11,2,3], and filters_grad_desc[3,3,5,3].
[       OK ] bad_indice_pairs_dim_shape_dtype/indice_convolution_backward_filter_general.negative/3 (0 ms)
[ RUN      ] bad_indice_pairs_dim_shape_dtype/indice_convolution_backward_filter_general.negative/4
[       OK ] bad_indice_pairs_dim_shape_dtype/indice_convolution_backward_filter_general.negative/4 (0 ms)
[ RUN      ] bad_indice_pairs_dim_shape_dtype/indice_convolution_backward_filter_general.negative/5
[2023-2-22 11:40:23] [MLUOP] [Error]:[mluOpGetIndiceConvolutionBackwardFilterWorkspaceSize] Shape check failed! Now the shapes are features_desc[3,5], output_grad_desc[3,3], indice_pairs_desc[9,1,3], and filters_grad_desc[3,3,5,3].
[       OK ] bad_indice_pairs_dim_shape_dtype/indice_convolution_backward_filter_general.negative/5 (0 ms)
[ RUN      ] bad_indice_pairs_dim_shape_dtype/indice_convolution_backward_filter_general.negative/6
[       OK ] bad_indice_pairs_dim_shape_dtype/indice_convolution_backward_filter_general.negative/6 (0 ms)
[ RUN      ] bad_indice_pairs_dim_shape_dtype/indice_convolution_backward_filter_general.negative/7
[2023-2-22 11:40:23] [MLUOP] [Error]:[mluOpGetIndiceConvolutionBackwardFilterWorkspaceSize] Shape check failed! Now the shapes are features_desc[3,5], output_grad_desc[3,3], indice_pairs_desc[9,2,3,4], and filters_grad_desc[3,3,5,3].
[       OK ] bad_indice_pairs_dim_shape_dtype/indice_convolution_backward_filter_general.negative/7 (0 ms)
[----------] 8 tests from bad_indice_pairs_dim_shape_dtype/indice_convolution_backward_filter_general (0 ms total)

[----------] 6 tests from bad_output_grad_dim_shape_dtype/indice_convolution_backward_filter_general
[ RUN      ] bad_output_grad_dim_shape_dtype/indice_convolution_backward_filter_general.negative/0
[       OK ] bad_output_grad_dim_shape_dtype/indice_convolution_backward_filter_general.negative/0 (0 ms)
[ RUN      ] bad_output_grad_dim_shape_dtype/indice_convolution_backward_filter_general.negative/1
[2023-2-22 11:40:23] [MLUOP] [Error]:[mluOpGetIndiceConvolutionBackwardFilterWorkspaceSize] The data type of features_desc, output_grad_desc and filters_grad_desc should be the same and the three should be either half or float. But now the data types are DTYPE_FLOAT-DTYPE_HALF-DTYPE_FLOAT.
[       OK ] bad_output_grad_dim_shape_dtype/indice_convolution_backward_filter_general.negative/1 (0 ms)
[ RUN      ] bad_output_grad_dim_shape_dtype/indice_convolution_backward_filter_general.negative/2
[       OK ] bad_output_grad_dim_shape_dtype/indice_convolution_backward_filter_general.negative/2 (0 ms)
[ RUN      ] bad_output_grad_dim_shape_dtype/indice_convolution_backward_filter_general.negative/3
[2023-2-22 11:40:23] [MLUOP] [Error]:[mluOpGetIndiceConvolutionBackwardFilterWorkspaceSize] Shape check failed! Now the shapes are features_desc[3,5], output_grad_desc[3,3,0], indice_pairs_desc[9,2,3], and filters_grad_desc[3,3,5,3].
[       OK ] bad_output_grad_dim_shape_dtype/indice_convolution_backward_filter_general.negative/3 (0 ms)
[ RUN      ] bad_output_grad_dim_shape_dtype/indice_convolution_backward_filter_general.negative/4
[       OK ] bad_output_grad_dim_shape_dtype/indice_convolution_backward_filter_general.negative/4 (0 ms)
[ RUN      ] bad_output_grad_dim_shape_dtype/indice_convolution_backward_filter_general.negative/5
[2023-2-22 11:40:23] [MLUOP] [Error]:[mluOpGetIndiceConvolutionBackwardFilterWorkspaceSize] Shape check failed! Now the shapes are features_desc[3,5], output_grad_desc[3,4], indice_pairs_desc[9,2,3], and filters_grad_desc[3,3,5,3].
[       OK ] bad_output_grad_dim_shape_dtype/indice_convolution_backward_filter_general.negative/5 (0 ms)
[----------] 6 tests from bad_output_grad_dim_shape_dtype/indice_convolution_backward_filter_general (0 ms total)

[----------] 8 tests from bad_filters_grad_dim_shape_dtype/indice_convolution_backward_filter_general
[ RUN      ] bad_filters_grad_dim_shape_dtype/indice_convolution_backward_filter_general.negative/0
[       OK ] bad_filters_grad_dim_shape_dtype/indice_convolution_backward_filter_general.negative/0 (0 ms)
[ RUN      ] bad_filters_grad_dim_shape_dtype/indice_convolution_backward_filter_general.negative/1
[2023-2-22 11:40:23] [MLUOP] [Error]:[mluOpGetIndiceConvolutionBackwardFilterWorkspaceSize] The data type of features_desc, output_grad_desc and filters_grad_desc should be the same and the three should be either half or float. But now the data types are DTYPE_FLOAT-DTYPE_FLOAT-DTYPE_HALF.
[       OK ] bad_filters_grad_dim_shape_dtype/indice_convolution_backward_filter_general.negative/1 (0 ms)
[ RUN      ] bad_filters_grad_dim_shape_dtype/indice_convolution_backward_filter_general.negative/2
[       OK ] bad_filters_grad_dim_shape_dtype/indice_convolution_backward_filter_general.negative/2 (1 ms)
[ RUN      ] bad_filters_grad_dim_shape_dtype/indice_convolution_backward_filter_general.negative/3
[2023-2-22 11:40:23] [MLUOP] [Error]:[mluOpGetIndiceConvolutionBackwardFilterWorkspaceSize] Shape check failed! Now the shapes are features_desc[3,5], output_grad_desc[3,3], indice_pairs_desc[9,2,3], and filters_grad_desc[3,3,5].
[       OK ] bad_filters_grad_dim_shape_dtype/indice_convolution_backward_filter_general.negative/3 (0 ms)
[ RUN      ] bad_filters_grad_dim_shape_dtype/indice_convolution_backward_filter_general.negative/4
[       OK ] bad_filters_grad_dim_shape_dtype/indice_convolution_backward_filter_general.negative/4 (0 ms)
[ RUN      ] bad_filters_grad_dim_shape_dtype/indice_convolution_backward_filter_general.negative/5
[2023-2-22 11:40:23] [MLUOP] [Error]:[mluOpGetIndiceConvolutionBackwardFilterWorkspaceSize] Shape check failed! Now the shapes are features_desc[3,5], output_grad_desc[3,3], indice_pairs_desc[9,2,3], and filters_grad_desc[3,3,15,3].
[       OK ] bad_filters_grad_dim_shape_dtype/indice_convolution_backward_filter_general.negative/5 (0 ms)
[ RUN      ] bad_filters_grad_dim_shape_dtype/indice_convolution_backward_filter_general.negative/6
[       OK ] bad_filters_grad_dim_shape_dtype/indice_convolution_backward_filter_general.negative/6 (0 ms)
[ RUN      ] bad_filters_grad_dim_shape_dtype/indice_convolution_backward_filter_general.negative/7
[2023-2-22 11:40:23] [MLUOP] [Error]:[mluOpGetIndiceConvolutionBackwardFilterWorkspaceSize] Shape check failed! Now the shapes are features_desc[3,5], output_grad_desc[3,3], indice_pairs_desc[9,2,3], and filters_grad_desc[3,3,5,13].
[       OK ] bad_filters_grad_dim_shape_dtype/indice_convolution_backward_filter_general.negative/7 (0 ms)
[----------] 8 tests from bad_filters_grad_dim_shape_dtype/indice_convolution_backward_filter_general (1 ms total)

[----------] 2 tests from bad_inverse/indice_convolution_backward_filter_general
[ RUN      ] bad_inverse/indice_convolution_backward_filter_general.negative/0
[       OK ] bad_inverse/indice_convolution_backward_filter_general.negative/0 (0 ms)
[ RUN      ] bad_inverse/indice_convolution_backward_filter_general.negative/1
[2023-2-22 11:40:23] [MLUOP] [Error]:[mluOpGetIndiceConvolutionBackwardFilterWorkspaceSize] Check failed: inverse == 0. 
[       OK ] bad_inverse/indice_convolution_backward_filter_general.negative/1 (0 ms)
[----------] 2 tests from bad_inverse/indice_convolution_backward_filter_general (0 ms total)

[----------] 1 test from bad_large_tensor_features/indice_convolution_backward_filter_general
[ RUN      ] bad_large_tensor_features/indice_convolution_backward_filter_general.negative/0
[2023-2-22 11:40:23] [MLUOP] [Warning]:[mluOpSetTensorDescriptor]: overflow max tensor num.Currently, mluOp supports tensor num smaller than 2^31,now tensor dims:(1147483648,5), data_width:4.
[2023-2-22 11:40:23] [MLUOP] [Error]:[mluOpGetIndiceConvolutionBackwardFilterWorkspaceSize] Overflow max tensor num. Currently, MLU-OPS supports tensor num smaller than 2^31.
[       OK ] bad_large_tensor_features/indice_convolution_backward_filter_general.negative/0 (0 ms)
[----------] 1 test from bad_large_tensor_features/indice_convolution_backward_filter_general (0 ms total)

[----------] 1 test from bad_large_tensor_output_grad/indice_convolution_backward_filter_general
[ RUN      ] bad_large_tensor_output_grad/indice_convolution_backward_filter_general.negative/0
[2023-2-22 11:40:23] [MLUOP] [Warning]:[mluOpSetTensorDescriptor]: overflow max tensor num.Currently, mluOp supports tensor num smaller than 2^31,now tensor dims:(715827883,3), data_width:4.
[2023-2-22 11:40:23] [MLUOP] [Error]:[mluOpGetIndiceConvolutionBackwardFilterWorkspaceSize] Overflow max tensor num. Currently, MLU-OPS supports tensor num smaller than 2^31.
[       OK ] bad_large_tensor_output_grad/indice_convolution_backward_filter_general.negative/0 (0 ms)
[----------] 1 test from bad_large_tensor_output_grad/indice_convolution_backward_filter_general (0 ms total)

[----------] 1 test from bad_large_tensor_indice_pairs/indice_convolution_backward_filter_general
[ RUN      ] bad_large_tensor_indice_pairs/indice_convolution_backward_filter_general.negative/0
[2023-2-22 11:40:23] [MLUOP] [Warning]:[mluOpSetTensorDescriptor]: overflow max tensor num.Currently, mluOp supports tensor num smaller than 2^31,now tensor dims:(715827883,2,3), data_width:4.
[2023-2-22 11:40:23] [MLUOP] [Error]:[mluOpGetIndiceConvolutionBackwardFilterWorkspaceSize] Overflow max tensor num. Currently, MLU-OPS supports tensor num smaller than 2^31.
[       OK ] bad_large_tensor_indice_pairs/indice_convolution_backward_filter_general.negative/0 (0 ms)
[----------] 1 test from bad_large_tensor_indice_pairs/indice_convolution_backward_filter_general (0 ms total)

[----------] 1 test from bad_large_tensor_filters_grad/indice_convolution_backward_filter_general
[ RUN      ] bad_large_tensor_filters_grad/indice_convolution_backward_filter_general.negative/0
[2023-2-22 11:40:23] [MLUOP] [Warning]:[mluOpSetTensorDescriptor]: overflow max tensor num.Currently, mluOp supports tensor num smaller than 2^31,now tensor dims:(47721859,3,5,3), data_width:4.
[2023-2-22 11:40:23] [MLUOP] [Error]:[mluOpGetIndiceConvolutionBackwardFilterWorkspaceSize] Overflow max tensor num. Currently, MLU-OPS supports tensor num smaller than 2^31.
[       OK ] bad_large_tensor_filters_grad/indice_convolution_backward_filter_general.negative/0 (0 ms)
[----------] 1 test from bad_large_tensor_filters_grad/indice_convolution_backward_filter_general (0 ms total)

```

**Review by alex-xuwenming — APPROVED** [review]

**Review by LittleReal — APPROVED** [review]
LGTM

**Pull request merged**